BuildLinux.sh

```diff
@@ -21,238 +19,172 @@ function check_available_memory_and_disk() {
-# libwebkit2gtk-4.1-dev ??
-export DEV_PACKAGES_COUNT=$(echo ${REQUIRED_DEV_PACKAGES} | wc -w)
-if [ $(dpkg --get-selections | grep -E "$(echo ${REQUIRED_DEV_PACKAGES} | tr ' ' '|')" | wc -l) -lt ${DEV_PACKAGES_COUNT} ]; then
-    sudo apt install -y ${REQUIRED_DEV_PACKAGES} git cmake wget file gettext
-fi
-
-#FIXME: require root for -u option
-if [[ -n "$UPDATE_LIB" ]]
+DISTRIBUTION=$(awk -F= '/^ID=/ {print $2}' /etc/os-release)
+if [ ! -f ./linux.d/${DISTRIBUTION} ]
 then
-    echo -n -e "Updating linux ...\n"
-    # hwclock -s # DeftDawg: Why does SuperSlicer want to do this?
-    apt update
-    if [[ -z "$FOUND_GTK3" ]]
-    then
-        echo -e "\nInstalling: libgtk2.0-dev libglew-dev libudev-dev libdbus-1-dev cmake git\n"
-        apt install -y libgtk2.0-dev libglew-dev libudev-dev libdbus-1-dev cmake git
-    else
-        echo -e "\nFind libgtk-3, installing: libgtk-3-dev libglew-dev libudev-dev libdbus-1-dev cmake git\n"
-        apt install -y libgtk-3-dev libglew-dev libudev-dev libdbus-1-dev cmake git
-    fi
-    # for ubuntu 22+ and 23+:
-    ubu_major_version="$(grep VERSION_ID /etc/os-release | cut -d "=" -f 2 | cut -d "." -f 1 | tr -d /\"/)"
-    if [ $ubu_major_version == "22" ] || [ $ubu_major_version == "23" ]
-    then
-        apt install -y curl libfuse-dev libssl-dev libcurl4-openssl-dev m4
-    fi
-    if [[ -n "$BUILD_DEBUG" ]]
-    then
-        echo -e "\nInstalling: libssl-dev libcurl4-openssl-dev\n"
-        apt install -y libssl-dev libcurl4-openssl-dev
-    fi
-    echo -e "done\n"
-    exit 0
+    echo "Your distribution does not appear to be currently supported by these build scripts"
+    exit 1
 fi
+source ./linux.d/${DISTRIBUTION}
 
-FOUND_GTK2_DEV=$(dpkg -l libgtk* | grep gtk2.0-dev || echo '')
-FOUND_GTK3_DEV=$(dpkg -l libgtk* | grep gtk-3-dev || echo '')
-echo "FOUND_GTK2=$FOUND_GTK2)"
-if [[ -z "$FOUND_GTK2_DEV" ]]
-then
-if [[ -z "$FOUND_GTK3_DEV" ]]
+echo "FOUND_GTK3=${FOUND_GTK3}"
+if [[ -z "${FOUND_GTK3_DEV}" ]]
 then
     echo "Error, you must install the dependencies before."
     echo "Use option -u with sudo"
     exit 0
 fi
-fi
 
-echo "[1/9] Updating submodules..."
-{
-    # update submodule profiles
-    pushd resources/profiles
-    git submodule update --init
-    popd
-}
-
-echo "[2/9] Changing date in version..."
+echo "Changing date in version..."
 {
     # change date in version
     sed -i "s/+UNKNOWN/_$(date '+%F')/" version.inc
 }
 echo "done"
 
-# mkdir in deps
-if [ ! -d "deps/build" ]
-then
-    mkdir deps/build
-fi
 
-if ! [[ -n "$SKIP_RAM_CHECK" ]]
+if ! [[ -n "${SKIP_RAM_CHECK}" ]]
 then
-check_available_memory_and_disk
+    check_available_memory_and_disk
 fi
 
-if [[ -n "$BUILD_DEPS" ]]
+if [[ -n "${BUILD_DEPS}" ]]
 then
-    echo "[3/9] Configuring dependencies..."
+    echo "Configuring dependencies..."
     BUILD_ARGS=""
-    if [[ -n "$FOUND_GTK3_DEV" ]]
+    if [[ -n "${FOUND_GTK3_DEV}" ]]
```

Comment:
remove this if statement since check is done above and it fails the program if not found